python/cudf/io/csv.py

```diff
@@ -77,21 +92,17 @@ def read_csv(filepath, lineterminator='\n',
     1 30   70
     """
 
-    if names is None or dtype is None:
-        msg = '''Automatic dtype detection not implemented:
-        Column names and dtypes must be specified.'''
-        raise TypeError(msg)
-
-    if isinstance(dtype, dict):
-        dtype_dict = True
-    elif isinstance(dtype, list):
-        dtype_dict = False
-        if len(dtype) != len(names):
+    if dtype is not None:
+        if isinstance(dtype, dict):
+            dtype_dict = True
+        elif isinstance(dtype, list):
```

Comment:
Should use `collections.abc.Iterable` instead of `list` to allow for tuples and other list-like objects.